rastervision_core/rastervision/core/data/dataset_config.py

```diff
@@ -14,6 +17,9 @@ class DatasetConfig(Config):
     validation_scenes: List[SceneConfig]
     test_scenes: List[SceneConfig] = []
 
+    img_channels: Optional[PositiveInt] = Field(
+        None, description='The number of channels of the images.')
```

Comment:
This broke backward compatibility with a model bundle from the 0.12 release. Anytime we add a new option with a *required value* to a config, or change the name of an option, we need to use a config upgrader to maintain backward compatibility for model bundles. Or, if it seems reasonable, you can add a default value and you don't need an upgrader.

You can see an example of adding a config version upgrader here: https://github.com/azavea/raster-vision/commit/9806a48dbeaea9972130eaeb1702f831f6a83f39 If you need more info on how the config upgrader works, we can have a quick meeting.

To test this kind of thing, I've been manually running the predict command with models from the Model Zoo https://docs.rastervision.io/en/latest/examples.html#model-zoo But, there is an issue to add this to the integration tests which we might want to prioritize now. https://github.com/azavea/raster-vision/issues/566

```
root@85d5380e10ec:/opt/src# rastervision predict https://s3.amazonaws.com/azavea-research-public-data/raster-vision/examples/model-zoo-0.12/isprs-potsdam-ss/model-bundle.zip \
>      https://s3.amazonaws.com/azavea-research-public-data/raster-vision/examples/model-zoo-0.12/isprs-potsdam-ss/3_12_sample.tif \
>      /opt/data/examples/spacenet-vegas-roads-ss/out.tif
Traceback (most recent call last):
  File "/opt/conda/lib/python3.6/runpy.py", line 193, in _run_module_as_main
    "__main__", mod_spec)
  File "/opt/conda/lib/python3.6/runpy.py", line 85, in _run_code
    exec(code, run_globals)
  File "/opt/src/rastervision_pipeline/rastervision/pipeline/cli.py", line 248, in <module>
    main()
  File "/opt/conda/lib/python3.6/site-packages/click/core.py", line 722, in __call__
    return self.main(*args, **kwargs)
  File "/opt/conda/lib/python3.6/site-packages/click/core.py", line 697, in main
    rv = self.invoke(ctx)
  File "/opt/conda/lib/python3.6/site-packages/click/core.py", line 1066, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
  File "/opt/conda/lib/python3.6/site-packages/click/core.py", line 895, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/opt/conda/lib/python3.6/site-packages/click/core.py", line 535, in invoke
    return callback(*args, **kwargs)
  File "/opt/src/rastervision_core/rastervision/core/cli.py", line 66, in predict
    predictor.predict([image_uri], output_uri)
  File "/opt/src/rastervision_core/rastervision/core/predictor.py", line 101, in predict
    self.pipeline.predict()
  File "/opt/src/rastervision_core/rastervision/core/rv_pipeline/rv_pipeline.py", line 149, in predict
    self.backend = self.config.backend.build(self.config, self.tmp_dir)
  File "/opt/src/rastervision_pytorch_backend/rastervision/pytorch_backend/pytorch_semantic_segmentation_config.py", line 39, in build
    learner = self.get_learner_config(pipeline)
  File "/opt/src/rastervision_pytorch_backend/rastervision/pytorch_backend/pytorch_semantic_segmentation_config.py", line 25, in get_learner_config
    channel_display_groups=pipeline.channel_display_groups)
  File "pydantic/main.py", line 274, in pydantic.main.BaseModel.__init__
pydantic.error_wrappers.ValidationError: 1 validation error for SemanticSegmentationDataConfig
img_channels
  none is not an allowed value (type=type_error.none.not_allowed)
```

Reply:
A little more info: the `img_channels` field was added to `SemanticSegmentationDataConfig` and is required. But old model bundles don't supply that field since it didn't exist. So the config upgrader needs to upgrade the version from 0 to 1, and set a value of 3 for `img_channels`. Each plugin has its own config version (which is currently 0 for `pytorch_backend`) so this will need to be incremented in `pytorch_backend/__init__.py`

Reply:
I have set a default value of 3 instead of using an upgrader. It seems reasonable because prior to this PR, RV only used 3-channel images.

I tried testing on the model bundles from the model zoo, but they seem to have an incompatible config structure. In particular, they don't have a learner config.

Reply:
I just tested using a bundle from the model zoo, and it worked after the changes you made. 

```
rastervision predict https://s3.amazonaws.com/azavea-research-public-data/raster-vision/examples/model-zoo-0.12/isprs-potsdam-ss/model-bundle.zip \
     https://s3.amazonaws.com/azavea-research-public-data/raster-vision/examples/model-zoo-0.12/isprs-potsdam-ss/3_12_sample.tif \
     /opt/data/examples/isprs-potsdam/out.tif
```